File: 刷题/中国移动大数据项目/workspace_IDEA/cmcc_visual/src/main/java/cn/sheep/cmcc/utils/Jpools.java
package cn.sheep.cmcc.utils;

import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
import redis.clients.jedis.Jedis;
import redis.clients.jedis.JedisPool;

/**
 * redis数据库连接池访问类
 */
public class Jpools {
    private static Config load = ConfigFactory.load();
    private static GenericObjectPoolConfig poolConfig = new GenericObjectPoolConfig();
    private static JedisPool jedisPool = null;

    static {
        poolConfig.setMaxIdle(load.getInt("redis.pool.maxIdle"));
        poolConfig.setMaxTotal(load.getInt("redis.pool.maxActive"));
        jedisPool = new JedisPool(poolConfig, load.getString("redis.ip"), load.getInt("redis.port"));
    }

    public static Jedis getJedis() {
        Jedis jedis = jedisPool.getResource();
        jedis.select(load.getInt("redis.db.index"));
        return jedis;
    }
}

File: 刷题/中国移动大数据项目/workspace_IDEA/cmcc_monitor/src/main/scala/cn/sheep/cmcc/utils/Jpools.scala
package cn.sheep.cmcc.utils

import org.apache.commons.pool2.impl.GenericObjectPoolConfig
import redis.clients.jedis.JedisPool

/**
  * ZhangJunJie
  * 2018/10/16 14:32
  * Describe：
  **/
object Jpools {

  private val poolConfig = new GenericObjectPoolConfig()
  poolConfig.setMaxIdle(5)      //最大的空闲连接数，连接池中最大的空闲连接数，默认是8
  poolConfig.setMaxTotal(2000)  //只支持最大的连接数，连接池中最大的连接数，默认是8

  //连接池是私有的不能对外公开访问
  private lazy val jedisPool = new JedisPool(poolConfig, AppParams.redisHost)

  def getJedis={
    val jedis = jedisPool.getResource
    jedis.select(AppParams.selectDBIndex)
    jedis
  }
}
